<?xml version="1.0" encoding="utf-8" ?>
<launch>
    <arg name="os1_imu_port" default="7503" doc="port to which the sensor should send imu data"/>
    <arg name="os1_lidar_port" default="7502" doc="port to which the sensor should send lidar data"/>
    <arg name="os1_udp_dest" default="192.168.15.100" doc="hostname or IP where the sensor will send data packets"/>


    <node pkg="nodelet" type="nodelet" name="ouster_nodelet_manager" args="manager">
        <param name="num_worker_threads" value="32" />
    </node>

    <node pkg="swri_nodelet"  type="nodelet" name="ouster_driver" args="os1_driver/os1_driver ouster_nodelet_manager">
        <param name="lidar_mode" type="string" value="1024x10"/>
        <param name="replay" value="false"/>
        <param name="os1_hostname" value="192.168.15.210"/>
        <param name="os1_udp_destination" value="$(arg os1_udp_dest)"/>
        <param name="os1_lidar_port" value="$(arg os1_lidar_port)"/>
        <param name="os1_imu_port" value="$(arg os1_imu_port)"/>
        <param name="use_ptp"   value="true"/>
    </node>

    <node pkg="swri_nodelet" type="nodelet" name="ouster_cloud" args="os1_driver/os1_packets_to_point_cloud ouster_nodelet_manager">
        <param name="tf_prefix" value="lidar_top" />
	    <param name="use_system_timestamp" value="true" />
    </node>
     
    <node pkg="swri_nodelet" type="nodelet" name="ouster_image" args="os1_driver/os1_point_cloud_to_image ouster_nodelet_manager" />

</launch>
